### PR TITLE
Band editor

### PR DESCRIPTION
**Pull request by thmasjs**
# Changes

*Describe the changes made, including any changes to tests, dependencies or docs. Link to any relevant issues below.*

- Closes #XXXX *(replace XXXX with GitHub issue number)*
- ...

# Tasks

- [ ] Have unittests been added (testing individual functions and their edge cases)
- [ ] Has documentation (including user guide) been updated
- [ ] Have new dependencies been added (or changed) in relevant `requirements.txt`
- [ ] Code linting, tests and other workflows pass

### CODE REVIEW

File: packages/vis-core/src/Components/Sidebar/Selectors/BandEditor.jsx [hunk 1]
```diff
@@ -1,78 +1,320 @@
...
+  background-color: #ffebee;
+  border-radius: 4px;
+  border-left: 3px solid #d32f2f;
 `;
 
-const AddRemoveButton = styled.button`
-  margin-left: 8px;
-  padding: 2px 8px;
-  font-size: 1em;
+const UpdateButton = styled.button`
```
Comment (brereh) [inline packages/vis-core/src/Components/Sidebar/Selectors/BandEditor.jsx]: Should use AppButton component for styling consistency

File: packages/vis-core/src/Components/Sidebar/Selectors/BandEditor.jsx [hunk 2]
```diff
@@ -1,78 +1,320 @@
...
 
-  const handleAddBand = () => {
-    setLocalBands([...localBands, localBands[localBands.length - 1] + 1]);
+  /**
+   * Handles changes to the number of bands.
+   * Adds new bands with incremental values or truncates existing bands.
+   *
+   * @param {Event} e - Change event from the select dropdown.
+   */
+  const handleBandCountChange = (e) => {
```
Comment (brereh) [inline packages/vis-core/src/Components/Sidebar/Selectors/BandEditor.jsx]: We should automatically reclassify the data on band count change, so that it's automatically captured. Otherwise it's more of a task to spread the data as you need to remember the parameters.

Ideally, you'd change number of bands, then it'd autospread. E.g. 9 --> 2, slice out the middle of the list to leave only the first and last indices, etc.

File: packages/vis-core/src/Components/Sidebar/Selectors/BandEditor.jsx [hunk 3]
```diff
@@ -1,78 +1,320 @@
...
+        <div style={{ display: 'flex', gap: 8 }}>
+          <BandCountSelect
+            value={localBands.length}
+            onChange={handleBandCountChange}
+          >
+            {Array.from({ length: maxBands - minBands + 1 }, (_, i) => minBands + i).map((n) => (
+              <option key={n} value={n}>{n}</option>
+            ))}
+          </BandCountSelect>
+          <UpdateButton
```
Comment (brereh) [inline packages/vis-core/src/Components/Sidebar/Selectors/BandEditor.jsx]: I think the reset button should go next to the update button, as a secondary button (so, 1/3, on the right, lighter BG). The temptation is to change the number of bands then hit 'reset' in this implementation.

**Review by brereh — CHANGES_REQUESTED** [review]
Some styling/UX issues to address (see comments).

I'm getting two issues:
1. Band editor showing for a layer with no data
2. When clicking 'update banding', it's resetting to the default, rather than what I set it to.

**NB this is after a test merge with dev -- aside from a slight flash when updating bands, it works with the current code**. 
Means we need to reconcile why the merge with dev breaks it. Imagine it's related to a classification method change.

**Comment by thmasjs** [comment]
Have done those changes @brereh

**Review by brereh — CHANGES_REQUESTED** [review]
Much better with the changes. Final refinements:
- Reset bands should go to the right of Update banding; rename to update and reset; make update 2/3 and reset 1/3 or row. Try with the Reset default colour diminutive to the update (so perhaps inverted colour)
- Place Classification method in row above number of bands; use the same font style as number of bands (the grey) for classification method, just to show how they're related.
- Instead of 'default' showing for a custom, show 'custom' in the classification method

@thmasjs

**Comment by thmasjs** [comment]
Changes made, have also added a segment splitter for the layer control entry to make it look neater. Have also sorted the merge conflict and seems to be running fine.

**Review by brereh — CHANGES_REQUESTED** [review]
It's looking really good. There are two important issues to fix. 

1. If you change the value then click update, it works. Subsequently changing the value and clicking update resets it back to what it was after the first update.
2. The validation on the values doesn't respect the fact that each band is the lower bound of the banding.

**Review by brereh — APPROVED** [review]
This is now working pretty well. There's one slight UX issue, in that if you change number of bands then hit reset, it'll go back to the old number. It's not a showstopper, so we are there with it.

Some actions to take forward:
reclassifyData: major refactoring of the whole function needed, to break it up and make it more SOLID. It works for now but is poorly maintainable because of
- repetition of logic
- multiple responsibilities
- multiple interfaces despite many only being used in one place ('d')
- breaks open-closed principle - open for extensibility, closed for modification

Add the ability to recalculate based on the new number of bands and the set classification method.

Both new issues to handle separately from here.
